server/channels/api4/channel.go

```diff
@@ -1875,6 +1877,77 @@ func updateChannelScheme(c *Context, w http.ResponseWriter, r *http.Request) {
 	ReturnStatusOK(w)
 }
 
+func getThreadsForChannel(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireChannelId().RequireUserId()
+	if c.Err != nil {
+		return
+	}
+
+	if !c.App.SessionHasPermissionToChannel(c.AppContext, *c.AppContext.Session(), c.Params.ChannelId, model.PermissionReadChannel) {
+		c.SetPermissionError(model.PermissionReadChannel)
+		return
+	}
+
+	opts := model.GetChannelThreadsOpts{
+		PageSize: uint64(c.Params.PerPage),
+	}
+
+	queryValues := r.URL.Query()
+
+	opts.Deleted, _ = strconv.ParseBool(queryValues.Get("deleted"))
+	opts.Since, _ = strconv.ParseUint(queryValues.Get("since"), 10, 64)
+	opts.Before = queryValues.Get("before")
+	opts.After = queryValues.Get("after")
+	opts.Extended, _ = strconv.ParseBool(queryValues.Get("extended"))
+
+	totalsOnlyKey := "totalsOnly"
+	if !queryValues.Has(totalsOnlyKey) {
+		totalsOnlyKey = "totals_only"
+	}
+	opts.TotalsOnly, _ = strconv.ParseBool(queryValues.Get(totalsOnlyKey))
+
+	threadsOnlyKey := "threadsOnly"
+	if !queryValues.Has(threadsOnlyKey) {
+		threadsOnlyKey = "threads_only"
+	}
+	opts.ThreadsOnly, _ = strconv.ParseBool(queryValues.Get(threadsOnlyKey))
+
+	// Before and After are mutually exclusive
+	if opts.Before != "" && opts.After != "" {
+		c.Err = model.NewAppError("Api4.getThreadsForChannel", "api.channel.get_threads_for_channel.invalid_before_after_params.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	// TotalsOnly and ThreadsOnly are mutually exclusive
+	if opts.TotalsOnly && opts.ThreadsOnly {
+		c.Err = model.NewAppError("Api4.getThreadsForChannel", "api.channel.get_threads_for_channel.invalid_totals_threads_params.app_error", nil, "", http.StatusBadRequest)
+		return
+	}
+
+	filter := queryValues.Get("filter")
+	if filter != "" {
+		switch model.GetChannelThreadsFilter(filter) {
+		case model.GetChannelThreadsFilterAll:
+			break
```

Comment:
It's because it's the same as sending an empty string or omitting the param. I've just kept it there so we know this is an acceptable value instead of passing an empty string to get all channel threads. 0/5